src/SFML/Graphics/Sprite.cpp

```diff
@@ -133,14 +129,11 @@ FloatRect Sprite::getGlobalBounds() const
 ////////////////////////////////////////////////////////////
 void Sprite::draw(RenderTarget& target, const RenderStates& states) const
 {
-    if (m_texture)
-    {
-        RenderStates statesCopy(states);
+    RenderStates statesCopy(states);
 
-        statesCopy.transform *= getTransform();
-        statesCopy.texture = m_texture;
-        target.draw(m_vertices, 4, PrimitiveType::TriangleStrip, statesCopy);
-    }
+    statesCopy.transform *= getTransform();
+    statesCopy.texture = m_texture;
+    target.draw(m_vertices, 4, PrimitiveType::TriangleStrip, statesCopy);
```

Comment:
Should we `assert(m_texture)`?

Reply:
Under what circumstances could that pointer become null? I used to have that check in there then removed it because as far as I can tell it's impossible for it to get to that state.

Reply:
> as far as I can tell it's impossible for it to get to that state

That seems a use case for `assert` :P

Reply:
`assert`s are so cheap that I won't protest its addition but I'm still really curious how under any circumstances other than `memcpy` shenanigans that pointer could become null.